tests/pack_pattern/README.rst

```diff
@@ -0,0 +1,36 @@
+Pack pattern annotation example
```

Comment:
Your test examples makes sense. **However**, I think v2x should be automatically detecting the requirement for a pack-pattern here and then generate these pack pattern names automatically.

A pack pattern is needed anytime there is an "internal pathway" which isn't connected to a top level port. IE
` [ ]->A->B->[ ]` - the internal signal `A->B` requires a pack pattern to be generated. I'm pretty sure we should be able to detect these internal only pathways?

In your example we have;
```
[ ] --> LUT --> MUX ------> [ ]
                 \
                  \-> FF -> [ ]
```

There are two "internal pathways", they are;
 * `LUT -> MUX` pathway.
 * `LUT -> MUX -> FF` pathway.

The names of the pack patterns can easily be generated as `<name_of_lut>_<name of mux>` and `<name_of_lut>_<name of mux>_<name of ff>`. (Or some other name.)

Reply:
The idea is sound. I'm trying to picture a scenario where it wouldn't be applicable.

Reply:
@mithro So I've done some thinking and I think that automatic pack-pattern inference won't work in general. I got some corner cases:

First of all V2X is not aware of the whole hierarchy of pb_types. It operates only on a given level at a time. This makes impossible to infer a pack pattern that would span multiple layers of the hierarchy. Such pack patterns will be required when modelling complex architectures like eg. IOB and I/OLOGIC in 7-series. Moreover V2X won't be able to generate a name for it as the name has to be unique throughout the hierarchy.

My second thought: What if there are multiple connections between cells? Specifying pack pattern on both of them would make it effective only when both connections are used. What if it is not the case?

Reply:
Can you post actual examples of these?